crates/matrix-sdk/src/authentication/oauth/registration_store.rs

```diff
@@ -164,8 +172,14 @@ impl OAuthRegistrationStore {
         });
         data.dynamic_registrations.insert(issuer, client_id);
 
-        let writer = BufWriter::new(File::create(&self.file_path)?);
-        serde_json::to_writer(writer, &data).map_err(OAuthRegistrationStoreError::IntoJson)
+        let file_path = self.file_path.clone();
+
+        spawn_blocking(move || {
+            let writer = BufWriter::new(File::create(&file_path)?);
+            serde_json::to_writer(writer, &data).map_err(OAuthRegistrationStoreError::IntoJson)
+        })
```

Comment:
Too bad `serde_json` doesn't support `AsyncWrite` here.

Reply:
We could also just serialize the whole file and then write it, if you'd prefer to use tokio APIs.

Reply:
Hmm that could be a good idea, yes. If it doesn't annoy you to change this part, it would be nice!

Reply:
I also prefer that, I just didn't want to change the code too much if it was some kind of necessary optimization to not have the whole file in memory.

Reply:
Let's change that as another commit then, no need to rebase or change your existing patches.

Reply:
Done